content/docs/base-components/ReactiveComponent.md

```diff
@@ -15,50 +15,110 @@ redirect_from:
+We can then use the given ReactiveComponent to be watched by the result component (via `react` prop) to avail the desired brand based filtering for the results.
 
-Check demo [here](https://codesandbox.io/s/3ylrrr0r5q).
+Check demo [here](https://codesandbox.io/s/7zrj740oj6).
 
 
 ```js
  <ReactiveComponent
-    componentId="CarSensor"
-    customQuery={props => ({
-        query: {
-                term: {
-                    "brand.keyword": "Ford"
-                }
-            }
+    componentId="Brand"
+    showFilter
+    defaultQuery={() => ({
+        aggs: {
+            'brand.keyword': {
+                terms: {
+                    field: 'brand.keyword',
+                    order: {
+                        _count: 'desc',
+                    },
+                    size: 5,
+                },
+            },
+        },
+    })}
+    render={data => {
+        return <TabComponent dataField="brand.keyword" {...data} />;
+    }}
+/>
+```
+
+You can also take advantage of various ReactiveSearch components static method for generating query. So we our basically selecting a single tab so we can use `SingleList` static method to generate query as follow:
+
+```js
+<ReactiveComponent
+    componentId="Brand"
+    showFilter
+    defaultQuery={() => ({
+        ...SingleList.generateQueryOptions({
+            dataField: "brand.keyword",
+            size: 5,
+            sortBy: "count"
         })
-    }
+    })}
+    render={data => {
+        return <TabComponent dataField="brand.keyword" {...data} />;
+    }}
 />
 ```
 
+Here `TabComponent` is the component responsible to render the Tabs UI and handle switching tab and displaying selected filters. We are using `render` prop to display the component which provides a object with different params that can used in rendering UI and dispatching new query.Fow switching tab we will use `onClick` event on each tab to fire new query. We will make use of following param provided by `render` method: 
+
+- **aggregations** `Object`
+    `aggregations` prop contains the results from `aggs` Elasticsearch query of the component.
+- **setQuery** `function`
+    `setQuery` function sets the query of the component. It takes an object param of shape:
+```javascript
+    {
+        query: {}, // query of the component
+        value: ''  // value of the component
+    }
+```
+- **value** `any`
+`value` contains the current value of the component (which can be set via `setQuery()` function). This is used for URLParams and SelectedFilters.
+
+**SelectedFilters with ReactiveComponent**      
+For showing selected filters we will use [`SelectedFilters`](/base-components/selectedfilters.html). When we use `ReactiveComponent` we will need to maintain `SelectedFilters` value manually. To clear the selected value of `ReactiveComponent` use `onClear` prop on `SelectedFilter` and pass `null` in `setQuery`.
```

Comment:
We should avoid assuming the user reading this doc is familiar with other parts of ReactiveSearch.